spring-data-jdbc/src/test/java/org/springframework/data/jdbc/core/JdbcAggregateTemplateIntegrationTests.java

```diff
@@ -466,7 +461,38 @@ public void saveAndLoadLongChain() {
 		template.delete(chain4, Chain4.class);
 
 		String countSelect = "SELECT COUNT(*) FROM %s";
-		jdbcTemplate.queryForObject(String.format(countSelect, "CHAIN0"),emptyMap(), Long.class);
+		jdbcTemplate.queryForObject(String.format(countSelect, "CHAIN0"), emptyMap(), Long.class);
+	}
+
+	@Test // DATAJDBC-359
+	public void saveAndLoadLongChainWithoutIds() {
+
+		NoIdChain4 chain4 = new NoIdChain4();
+		chain4.fourValue = "omega";
+		chain4.chain3 = new NoIdChain3();
+		chain4.chain3.threeValue = "delta";
+		chain4.chain3.chain2 = new NoIdChain2();
+		chain4.chain3.chain2.twoValue = "gamma";
+		chain4.chain3.chain2.chain1 = new NoIdChain1();
+		chain4.chain3.chain2.chain1.oneValue = "beta";
+		chain4.chain3.chain2.chain1.chain0 = new NoIdChain0();
+		chain4.chain3.chain2.chain1.chain0.zeroValue = "alpha";
+
+		template.save(chain4);
+
+		assertThat(chain4.four).isNotNull();
+
+		NoIdChain4 reloaded = template.findById(chain4.four, NoIdChain4.class);
+
+		assertThat(reloaded).isNotNull();
+
+		assertThat(reloaded.four).isEqualTo(chain4.four);
+		assertThat(reloaded.chain3.chain2.chain1.chain0.zeroValue).isEqualTo(chain4.chain3.chain2.chain1.chain0.zeroValue);
+
+		template.delete(chain4, NoIdChain4.class);
+
+		String countSelect = "SELECT COUNT(*) FROM %s";
```

Comment:
Why `String.format(…)` instead of having directly `SELECT COUNT(*) CHAIN0`? Also, the query result is not used/asserted.